src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs

```diff
@@ -122,9 +122,9 @@ public partial struct ImmutableArray<T> : System.Collections.Generic.ICollection
         public System.ReadOnlyMemory<T> AsMemory() { throw null; }
         public System.ReadOnlySpan<T> AsSpan() { throw null; }
 #endif
-        public System.Collections.Immutable.ImmutableArray<TOther> As<TOther>() where TOther : class { throw null; }
-        public System.Collections.Immutable.ImmutableArray<TOther> CastArray<TOther>() where TOther : class { throw null; }
-        public static System.Collections.Immutable.ImmutableArray<T> CastUp<TDerived>(System.Collections.Immutable.ImmutableArray<TDerived> items) where TDerived : class, T { throw null; }
+        public System.Collections.Immutable.ImmutableArray<TOther> As<TOther>() where TOther : class? { throw null; }
```

Comment:
NIT: This one also needs to be `?` ?
CC @333fred

Reply:
It seems like it would be perfectly reasonable to pass a nullable type here.

Reply:
In fact, it seems like this will be a place to use `T?` in the return type when that's available.

Reply:
I'm not sure I follow, what is the proposed alteration here?

Reply:
That is related to https://github.com/dotnet/runtime/issues/39877, no action needed for this PR